pkg/workload/workload.go

```diff
@@ -965,20 +965,34 @@ func HasConditionWithTypeAndReason(w *kueue.Workload, cond *metav1.Condition) bo
 	return false
 }
 
-func HasNodeToReplace(w *kueue.Workload) bool {
-	return w != nil && len(w.Status.NodesToReplace) > 0
+func HasUnhealthyNodes(w *kueue.Workload) bool {
+	return w != nil && len(w.Status.UnhealthyNodes) > 0
 }
 
-func HasTopologyAssignmentWithNodeToReplace(w *kueue.Workload) bool {
-	if !HasNodeToReplace(w) || !IsAdmitted(w) {
+func HasUnhealthyNode(w *kueue.Workload, nodeName string) bool {
+	return slices.ContainsFunc(w.Status.UnhealthyNodes, func(node kueue.UnhealthyNode) bool {
+		return node.Name == nodeName
+	})
+}
+
+func UnhealthyNodeNames(w *kueue.Workload) []string {
+	unhealthyNodeNames := make([]string, len(w.Status.UnhealthyNodes))
+	for i, unhealthyNode := range w.Status.UnhealthyNodes {
+		unhealthyNodeNames[i] = unhealthyNode.Name
+	}
+	return unhealthyNodeNames
```

Comment:
```suggestion
	return utilslices.Map(w.Status.UnheadlthyNodes, func(unhealthyNode *kueue.UnhealthyNode) string { return unhealthyNode.Name} )
```

Let's leverage our library.

Reply:
Done in https://github.com/kubernetes-sigs/kueue/pull/6822.